controller/model/posture_check_model_process.go

```diff
@@ -23,12 +23,41 @@ import (
 )
 
 type PostureCheckProcess struct {
+	PostureCheckId  string
 	OperatingSystem string
 	Path            string
 	Hashes          []string
 	Fingerprint     string
 }
 
+func (p *PostureCheckProcess) Evaluate(pd *PostureData) bool {
+	for _, process := range pd.Processes {
+		if process.PostureCheckId == p.PostureCheckId {
+			if process.TimedOut {
+				return false
+			}
+
+			if p.Fingerprint != "" {
+				if p.Fingerprint != process.SignerFingerprint {
+					return false
```

Comment:
shouldn't this just 'continue' and not return? it's in a for loop - don't you need to check them all before returning a response?

Reply:
If there is a signer fingerprint defined, and it is missing, the posture fails. There is no point in continuing.

Reply:
I think I see your confusion.

PostureData `pd` can have multiple processes reported from the client. In this evaluation, we are looking for the single reported process in `pd.Processes` by looking at the `PostureCheckId`. The loop should only hit that `PostureCheckId` comparison once and when it does it is ride or die, hence all the returns.